### ISSUE TITLE
Introduce semantic-release

### ISSUE THREAD

**Issue by czosel** [prettier/plugin-php#1404]
What do you think about adding [semantic-release](https://semantic-release.gitbook.io/semantic-release/) to the repository? It would mean we'd have to enforce [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
I like semantic release because it allows more frequent releases, automates release notes, and automatically adds a comment to issues and PRs when the release is available.

Edit: [possible workflow](https://github.com/semantic-release/semantic-release/blob/0785a844fa8ac1320383452ce531898be3b01f92/docs/recipes/distribution-channels.md)

Edit 2: Possible disadvantages:

- semantic release doesn't use the `0.x` version range. Out first release would probably trigger `1.0.0`. See [reasoning](https://semantic-release.gitbook.io/semantic-release/support/faq#can-i-set-the-initial-release-version-of-my-package-to-0-0-1).
- if we automatically release after every merge to `master` (default config of semantic-release), we might get more releases than we actually want. See [reasoning](https://semantic-release.gitbook.io/semantic-release/support/faq#is-it-really-a-good-idea-to-release-on-every-push).

Alternatives:

- only trigger releases when we merge into a dedicated branch, e.g. `next`
- don't use semantic release, just enforce conventional commits to automate the creation of release notes

cc @loilo @evilebottnawi

**Comment by loilo** [comment]
Semantic release is something I've always wanted to try but eventually never gave a shot because it wasn't worth the extra work on my scale of projects. Also I, personally, find it hard to distinguish between feature and bugfix more often than you'd think.

I also refrained from it because of the second disadvantage you're mentioning - I didn't like the thought of giving up control over releases. I'd tend to only use semantic release for creation of release nots, but then again - I'm not the one writing the release notes anyway, so it's mostly on you @czosel to decide how much of a hassle writing those release notes currently is.

**tl;dr:** I like the automation, but I'm not sure (not in disagreement, I'm really just not sure) if we are or ever will be at the scale to really need it.

**Comment by evilebottnawi** [comment]
Yes, semantic releases are great, but we really are not on such a scale, there are also some interesting thing - sometimes we do fixes/features using 2-3 PRs :smile: I think we can leave it open and return in the future.

**Comment by czosel** [comment]
Thanks for your feedback - you’re right, the longer I thought about it the more doubt I had as well. Probably just got excited when I considered doing a release for the PR we merged today 😉 I’ll go ahead and close this.
